.circleci/config.yml

```diff
@@ -41,10 +148,140 @@ jobs:
             - run:
                 name: coveralls
                 command: npm run coveralls
-      - save_cache:
-          key: dependency-cache-{{ .Environment.CIRCLE_JOB }}-{{ checksum "package.json" }}
-          paths:
-            - ./node_modules
+
+  hardhat-core-default-solc:
+    docker:
+      - image: circleci/node:16
+    steps:
+      - show_npm_version
+      - provision_and_package_solcjs
+      - provision_hardhat_with_packaged_solcjs
+      - run:
+          name: Run hardhat-core test suite with its default solc
+          command: |
+            cd hardhat/packages/hardhat-core
+            yarn test
```

Comment:
This test is **much** slower than `hardhat-core-latest-solc`. It runs with 0.5.x and runs about 4k tests, which takes ~15 min. `hardhat-core-latest-solc` has 10 times fewer tests and finishes in ~1 min.

Not sure if we should keep it or not. I think it may still be adding some value because I think that tests on 0.8.x do not actually go through all the features. In particular Hardhat's downloader module tests get skipped and that's one of the few things in Hardhat that uses solc-js.

I think that Hardhat prepared `HARDHAT_TESTS_SOLC_PATH` specifically with testing the latest compiler in mind and disabled the extra tests on purpose. Probably because they're not relevant if you're testing solc, but I think some of them do matter for solc-js.

Reply:
Since we don't have a really high volume of changes in solc-js, I guess 15 minutes tests is not *that* bad...